packages/app/src/files-manager/components/duplicated-files-modal/duplicated-files-modal.component.tsx

```diff
@@ -0,0 +1,163 @@
+
+const messages = defineMessages({
+  duplicatedFilesUploadModalTitle: {
+    id: "module.filesManager.duplicatedFilesUploadModalTitle",
+  },
+  duplicatedFilesUploadModalHeader: {
+    id: "module.filesManager.duplicatedFilesUploadModalHeader",
+  },
+  duplicatedFilesUploadModalTextInfo: {
+    id: "module.filesManager.duplicatedFilesUploadModalTextInfo",
+  },
+  duplicatedFilesUploadModalPendingFilesTextInfo: {
+    id: "module.filesManager.duplicatedFilesUploadModalPendingFilesTextInfo",
+  },
+  duplicatedFilesUploadModalActionButton: {
+    id: "module.filesManager.duplicatedFilesUploadModalActionButton",
+  },
+  duplicatedFilesUploadModalListTitle: {
+    id: "module.filesManager.duplicatedFilesUploadModalListTitle",
+  },
+})
+
+const DuplicatedFilesDetailText = styled(Text)`
+  margin-top: 0.8rem;
+`
+
+const DuplicatedFilesListWrapper = styled.div`
+  width: 100%;
+  margin-top: 2.4rem;
+`
+const DuplicatedFilesListItems = styled.div`
+  max-height: 10rem;
+  overflow: auto;
+`
+
+const DuplicatedFilesListTitleText = styled(Text)`
+  margin-bottom: 1.1rem;
+  border-width: 0 0 0.1rem 0;
+  border-style: solid;
+  border-color: ${({ theme }: ThemeProps<Theme>) => {
+    return theme.color.border.verticalSeparator
+  }};
+  text-align: left !important;
+`
+
+const DuplicatedFilesListItemText = styled(Text)`
+  text-align: left !important;
+`
+
+const DuplicatedFilesModal: FunctionComponent = ({ ...props }) => {
+  const filesManagerError = useSelector(getFilesManagerError)
+
+  const uploadPendingFiles = useSelector(getUploadPendingFiles)
+  const duplicatedFiles = useSelector(getDuplicatedFiles)
+
+  const dispatch = useDispatch()
+
+  if (filesManagerError?.type !== FilesManagerError.UploadDuplicates) {
+    return null
+  }
+
+  const detailText =
+    uploadPendingFiles.length > 0
+      ? {
+          ...messages.duplicatedFilesUploadModalPendingFilesTextInfo,
+          values: {
+            uploadFilesCount:
+              duplicatedFiles.length + uploadPendingFiles.length,
+            duplicatedFilesCount: duplicatedFiles.length,
+          },
+        }
+      : messages.duplicatedFilesUploadModalTextInfo
+
+  const modalDialogSize =
+    uploadPendingFiles.length > 0 ? ModalSize.Medium : ModalSize.Small
+
+  const duplicatedFilesList =
+    uploadPendingFiles.length > 0 ? (
+      <DuplicatedFilesListWrapper>
+        <DuplicatedFilesListTitleText
+          displayStyle={TextDisplayStyle.Paragraph4}
+          color="secondary"
+          message={messages.duplicatedFilesUploadModalListTitle}
+        />
+        <DuplicatedFilesListItems>
+          {duplicatedFiles.map((fileName, index) => {
+            return (
+              <DuplicatedFilesListItemText
+                displayStyle={TextDisplayStyle.Paragraph4}
+                color="secondary"
+                message={fileName}
+                key={`duplicated-files-list-item-text-${index}`}
```

Comment:
I don't think that indexes are great keys, the file name would be a better choice i guess

Reply:
resolved in c93e17353a55ac2e1b9b8b9c4ea45d46e986cbdc